catalogue_graph/src/utils/bulk_load.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+
+from pydantic import BaseModel
+
+import config
+from utils.types import (
+    EntityType,
+    TransformerType,
+)
+
+
+class IncrementalWindow(BaseModel):
+    start_time: datetime
+    end_time: datetime
+
+
+def get_file_path(
+    transformer_type: TransformerType,
+    entity_type: EntityType,
+    pipeline_date: str,
+    window: IncrementalWindow | None = None,
+) -> str:
+    """Return the file path of a bulk load file."""
+    file_name = f"{transformer_type}__{entity_type}.csv"
+
+    window_prefix = ""
+    if window is not None:
+        start = window.start_time.strftime("%Y%m%dT%H%M")
+        end = window.end_time.strftime("%Y%m%dT%H%M")
+        window_prefix = f"windows/{start}-{end}/"
+
+    return f"{pipeline_date}/{window_prefix}{file_name}"
```

Comment:
```suggestion
from pathlib import PurePosixPath

def get_file_path(
    transformer_type: TransformerType,
    entity_type: EntityType,
    pipeline_date: str,
    window: IncrementalWindow | None = None,
) -> str:
    """Return the file path (object key) of a bulk load file."""
    file_name = f"{transformer_type}__{entity_type}.csv"

    parts: list[str] = [pipeline_date]
    if window is not None:
        start = window.start_time.strftime("%Y%m%dT%H%M")
        end = window.end_time.strftime("%Y%m%dT%H%M")
        parts += ["windows", f"{start}-{end}"]

    file_path = PurePosixPath(*parts) / file_name
    return str(file_path)
```

Maybe it would be preferable to use `pathlib` to avoid the string substitution construction of paths?